Hadrons/Modules/MDistil/DistilMatrix.hpp

```diff
@@ -28,20 +28,53 @@ class DistilMesonFieldMetadata: Serializable
                                     unsigned int,               Nt,
                                     unsigned int,               Nvec,
                                     std::vector<RealF>,         Momentum,
-                                    Gamma::Algebra,             Opr,               // just gamma matrices for now, but could turn into more general operators in the future
+                                    Gamma::Algebra,             Operator,               // just gamma matrices for now, but could turn into more general operators in the future
                                     std::vector<int>,           NoisePair,
-                                    std::string,                MesonFieldType)
+                                    std::string,                MesonFieldType,
+                                    std::string,                NoiseHashLeft,
+                                    std::string,                NoiseHashRight,)
 };
 
-// class TimeDilutionBlocksMetadata: Serializable
-// {
-// public:
-//     GRID_SERIALIZABLE_CLASS_MEMBERS(TimeDilutionBlocksMetadata,
-//                                     std::vector<unsigned int>,  left_time_sources,
-//                                     std::vector<unsigned int>,  right_time_sources,
-//                                     TimeSliceMap,               left_time_dilution,
-//                                     TimeSliceMap,               right_time_dilution)
-// };
+//metadata io class
+class DistilMetadataIo
+{
+private:
+    std::string filename_, metadataname_;
+public:
+    // constructor
+    DistilMetadataIo(std::string filename, std::string metadataname):filename_(filename) , metadataname_(metadataname) {}
+    //methods
+    template <typename T>
+    void write2dMetadata(const std::string name, const std::vector<std::vector<T>>& data)
+    {
+        // auxiliar variable-length struct (see hdf5 variable-length documentation)
+        typedef struct  {
+            size_t len; /* Length of VL data (in base type units) */      
+            void *p;    /* Pointer to VL data */        
+        } VlStorage;
+#ifdef HAVE_HDF5
+
+        Hdf5Reader  reader(filename_, false);
```

Comment:
Why use HDF5Reader when you write in this routine?

Reply:
It's just to get the reference to the h5 group, I can change to Hdf5Writer without consequence (next push)

Reply:
Sorry, that's not true. I use `reader` because the group is already created at this point. If I use a `writer` grid will try to create a group in the file when using `push()`, so I need to use a reader with `readOnly=false` just to get the reference to the group.